File: Data Structures/BinarySearch.java
/*Sample Input
Enter number of elements
5
Enter 5 integers
78
67
89
56
45
Enter a number to search for: 67
Sample Output
Found at index: 1  */

import java.util.*;

public class BinarySearch {

    public static void main(String[] args) {

        int n, a[];
        int searchV = 0, index;

        Scanner input = new Scanner(System.in);
        System.out.println("Enter number of elements");
        n = input.nextInt();
        a = new int[n];
        System.out.println("Enter " + n + " integers");
        // now we put all the value in a previous define array with help of scanner
        // class
        for (int i = 0; i < n; i++) {
            a[i] = input.nextInt();
        }
        System.out.print("Enter a number to search for: ");
        // take the user input which he wants to search
        searchV = input.nextInt();
        index = binarySearch(a, searchV);
        if (index != -1) {
            System.out.println("Found at index: " + index);
        } else {
            System.out.println("Not Found");
        }
    }

    // here we define binary search method
    static int binarySearch(int[] search, int find) {
        int start, end, midPt;
        start = 0;
        end = search.length - 1;
        while (start <= end) {
            midPt = (start + end) / 2;
            if (search[midPt] == find) {
                return midPt;
            } else if (search[midPt] < find) {
                start = midPt + 1;
            } else {
                end = midPt - 1;
            }
        }
        return -1;
    }
}

File: Algorithms/Binary_search.py

value=[1,2,5,7,8,9,11,20]
number = int(input("Enter the Number to be search "))
num_val=len(value)-1

def binary_search(value,first,end,val):
    if end >= 1:
        mid = first + (end-1)//2
        print("\n")

        if value[mid] == val:
            return mid
        elif value[mid] >= val:
            return binary_search(value,first,mid-1,val)
        elif value[mid] <= val:
            return binary_search(value, mid +1,end ,val)


result = binary_search(value,0,num_val,number)

if result != -1:
    print(number," is found at ", result ," position")
else :
    print(" No found ")
'''
  Test case:-
   number = 20
   1st mid = 3 , then value[3] = 7
                   since 7< 20
                   therefore first = mid + 1 = 4
                              
   2nd mid = 7 , then value[7] = 20
                    value = 20
                    Number is found

'''
